commit abb04663edbbdb6d878fce7ecd0fff1eb192ad70
Author: Sergei Trofimovich <slyfox@gentoo.org>
Date:   Sun May 4 10:50:10 2014 +0300

    fix build failure against ghc-HEAD (die clast, FlexibleContexts)
    
    The name clash failure is easy:
    >    Ambiguous occurrence ‘die’
    >    It could refer to either ‘Main.die’, defined at src/Main.lhs:291:3
    >                          or ‘System.Exit.die’,
    >                             imported from ‘System.Exit’ at src/Main.lhs:28:3-20
    
    The type inference constraints with language extensions are
    more tight now:
    http://permalink.gmane.org/gmane.comp.lang.haskell.glasgow.user/24612
    
    Signed-off-by: Sergei Trofimovich <slyfox@gentoo.org>

diff --git a/happy.cabal b/happy.cabal
index b0b5e2a..da58632 100644
--- a/happy.cabal
+++ b/happy.cabal
@@ -138,7 +138,7 @@ executable happy
 
   build-depends: base < 5, mtl >= 1.0
 
-  extensions: CPP, MagicHash
+  extensions: CPP, MagicHash, FlexibleContexts
   ghc-options: -Wall -fno-warn-type-defaults
   other-modules:
         AbsSyn
diff --git a/src/Main.lhs b/src/Main.lhs
index cfe2942..b318211 100644
--- a/src/Main.lhs
+++ b/src/Main.lhs
@@ -25,7 +25,7 @@ Path settings auto-generated by Cabal:
 > import System.Console.GetOpt
 > import Control.Monad ( liftM )
 > import System.Environment
-> import System.Exit
+> import System.Exit (exitWith, ExitCode(..))
 > import Data.Char
 > import System.IO
 > import Data.Array( assocs, elems, (!) )
